src/QMCWaveFunctions/Fermion/SlaterDetBuilder.cpp

```diff
@@ -1218,4 +1244,194 @@ bool SlaterDetBuilder::readDetList(xmlNodePtr cur, std::vector<ci_configuration>
   return success;
 }
 
+
+
+bool SlaterDetBuilder::readDetListH5(xmlNodePtr cur, std::vector<ci_configuration>& uniqueConfg_up, std::vector<ci_configuration>& uniqueConfg_dn, std::vector<size_t>& C2node_up, std::vector<size_t>& C2node_dn, std::vector<std::string>& CItags, std::vector<RealType>& coeff, bool& optimizeCI, int nels_up, int nels_dn)
+{
+
+  bool success=true;
+  uniqueConfg_up.clear();
+  uniqueConfg_dn.clear();
+  C2node_up.clear();
+  C2node_dn.clear();
+  CItags.clear();
+  coeff.clear();
+  std::vector<ci_configuration> confgList_up;
+  std::vector<ci_configuration> confgList_dn;
+  std::vector<RealType> CIcoeff;
+  std::vector<std::string> ConfigTag;
+  std::string optCI="no";
+  RealType cutoff=0.0;
+  OhmmsAttributeSet ciAttrib;
+  ciAttrib.add (optCI,"optimize");
+  ciAttrib.put(cur);
+  optimizeCI = (optCI=="yes");
+  xmlNodePtr curRoot=cur,DetListNode;
+  std::string cname,cname0,multidetH5path;
+  cur = curRoot->children;
+  while (cur != NULL)//check the basis set
+  {
+    getNodeName(cname,cur);
+    if(cname == "detlist")
+    {
+      DetListNode=cur;
+      app_log() <<"Found determinant list. \n";
+    }
+    cur = cur->next;
+  }
+  size_t NCA,NCB,NEA,NEB,nstates,ndets=0,count=0,cnt0=0;
+  size_t H5_ndets,H5_nstates;
+  int N_int;
+  const int bit_kind = 64; 
+  std::string Dettype="DETS";
+  RealType sumsq=0.0;
+  OhmmsAttributeSet spoAttrib;
+  spoAttrib.add (NCA, "nca");
+  spoAttrib.add (NCB, "ncb");
+  spoAttrib.add (NEA, "nea");
+  spoAttrib.add (NEB, "neb");
+  spoAttrib.add (ndets, "size");
+  spoAttrib.add (Dettype, "type");
+  spoAttrib.add (nstates, "nstates");
+  spoAttrib.add (cutoff,"cutoff");
+  spoAttrib.add (multidetH5path,"href");
+  spoAttrib.put(DetListNode);
+  if(ndets==0)
+  {
+    APP_ABORT("size==0 in detlist is not allowed. Use slaterdeterminant in this case.\n");
+  }
+  if(Dettype != "DETS" && Dettype != "Determinants")
```

Comment:
Just spot this by change. More for discussion, not asking for a change in this PR.
Is it better to support only Determinants in the future? I don't see any benefit keeping both except supporting legacy.
@anbenali Does the converter only write DETS or Determinants?

Reply:
this is for the case of CSF.  Since I do not use CSFs and the only way (for now) to generate a CSF is through Gamess and I will not touch gamess unless Paul Kent forces me to do so I only focused on what is needed to do science.
This being said, CSFs are important and should still be supported. Also, it is unlikely to endup with millions of determinants in CSF so the XML path is fine.

Reply:
Yes. I mean to keep minimal amount necessary input tags. We probably should only keep CSF and Determinants and deprecate all others by stopping the code.